File: sdks/python/src/bigrag/resources/admin/vector_storage.py
from __future__ import annotations

from typing import TYPE_CHECKING

from bigrag.types.admin import VectorStorageOverviewResponse

if TYPE_CHECKING:
    from bigrag._core import BigRAGCore


class AdminVectorStorageResource:
    def __init__(self, client: BigRAGCore) -> None:
        self._client = client

    async def overview(self) -> VectorStorageOverviewResponse:
        return await self._client._request("GET", "/v1/admin/vector-storage/overview")

File: sdks/typescript/src/resources/admin/vector_storage.ts
import type { RequestClient } from "../../core.js";
import type { VectorStorageOverviewResponse } from "../../types/index.js";

export class AdminVectorStorageResource {
  constructor(private readonly _client: RequestClient) {}

  overview(): Promise<VectorStorageOverviewResponse> {
    return this._client._request("GET", "/v1/admin/vector-storage/overview");
  }
}
